vcf_view.wdl

```diff
@@ -40,8 +41,16 @@ task xVCFView {
         Int cpu = 8
         Int memory = 64
         Int preemptible = 0
+        Int addl_disk = 1
     }
+    # estimate disk size required
+    Int vcf_size        = ceil(size(input_vcf, "GB"))
+    Int index_size      = select_first([ceil(size(input_vcf_index, "GB")), 0])
+    Int samples_size    = select_first([ceil(size(samples, "GB")), 0])
+    Int regions_size    = select_first([ceil(size(regions, "GB")), 0])
```

Comment:
I've never written WDL before, so excuse my ignorance, but what does `select_first()` do here? Is this similar to finding the max of the size and 0?

Reply:
select_first() is a built-in WDL function that selects the leftmost (ie, first) entry in its input array that is defined.

Example: input_vcf_index is an optional input. By extension, so is size(input_vcf_index, "GB"). If input_vcf_index is defined, then size(input_vcf_index, "GB") is defined, and the value of index_size will be set to size(input_vcf_index, "GB"). However, if input_vcf_index is not defined, then size(input_vcf_index, "GB") will not be defined, so select_first will look at the next value. In this case that is 0, so the value of index_size will be 0.